### ISSUE TITLE
Add built-in support for converting "->" to "→"

### ISSUE THREAD

**Issue by lamtuanamc** [TypeCellOS/BlockNote#915]
Hi,

I would like to suggest adding built-in support for automatically converting "->" to "→" when typing.

This feature would be very useful for many users who often use arrow symbols in their texts.

Thanks for considering this request!

**Comment by matthewlipski** [comment]
There's not a good way to do this directly from the BlockNote API atm unfortunately, but should be quite easy with a [TipTap extension](https://tiptap.dev/docs/editor/extensions/custom-extensions/extend-existing#input-rules). You can then pass your extension to the `_tiptapOptions` editor option:

```ts
const editor = useCreateBlockNote({
  _tiptapOptions: {
    extensions: [extension];
  }
});
```

**Comment by lamtuanamc** [comment]
ok thanks, i got it
```
import { Extension } from '@tiptap/core'

const ArrowConversion = Extension.create({
  name: 'arrowConversion',

  addInputRules() {
    return [
      {
        find: /->/g,
        handler: ({ state, range }) => {
          const { from, to } = range
          const tr = state.tr.replaceWith(from, to, state.schema.text('→'))
          return tr
        },
      },
    ]
  },
})

export default ArrowConversion

```

**Comment by kom-senapati** [comment]
I would like to give it a try. Can you guide me how to setup the repo for development and where I have to change

**Comment by lamtuanamc** [comment]
> ok thanks, i got it
> 
> ```
> import { Extension } from '@tiptap/core'
> 
> const ArrowConversion = Extension.create({
>   name: 'arrowConversion',
> 
>   addInputRules() {
>     return [
>       {
>         find: /->/g,
>         handler: ({ state, range }) => {
>           const { from, to } = range
>           const tr = state.tr.replaceWith(from, to, state.schema.text('→'))
>           return tr
>         },
>       },
>     ]
>   },
> })
> 
> export default ArrowConversion
> ```

I see you used this code in the pull request. But I encountered some editing errors when using this code, when adding a space between words, or adding a few special characters.
If you can create a pull request to make an example of how to use the Tiptap extension, please check the logic carefully

@kom-senapati

**Comment by kom-senapati** [comment]
Can you tell me what editing errors you got?

**Comment by kom-senapati** [comment]
The extension is added to Blocknote, it works in playground.

**Comment by lamtuanamc** [comment]
I used the code snippet above in _tiptapOptions and encountered an error when adding a space to a word, or when modifying a character. It gets replaced by an arrow. When I tested it, there was no issue, but the tester reported this error when it was moved to production.

https://github.com/user-attachments/assets/496b229d-c50f-40e8-8278-ba842ba613e2

https://github.com/user-attachments/assets/0fc83cd7-7c24-4d35-952d-a47259260e8b
